ksqldb-engine/src/main/java/io/confluent/ksql/planner/plan/PullFilterNode.java

```diff
@@ -295,7 +297,23 @@ public Void visitComparisonExpression(
         final ComparisonExpression node,
         final Object context
     ) {
+      if (!isSingleColumnReference(node)) {
+        setTableScanOrElseThrow(() ->
+            invalidWhereClauseException("A comparison must reference a key column and literal",
+                isWindowed));
+        return null;
+      }
+
       final UnqualifiedColumnReferenceExp column = getColumnRefSide(node);
+      final Expression other = getNonColumnRefSide(node);
```

Comment:
nit: I'm wondering if we can merge these two conditions into a single one, like:

```
first = getFirstColumnRefSide(); // if both are column ref, return left; if non are column ref, return null;
if (first != null) {
    second = getOtherSide();
    secondHasColumnRef = new ... ; secondHasColumnRef.process(second);
    if (second isColumnRef || secondHasColumnRef.hasColumnRef)
        setTableScan();
} else {
    setTableScan();    // i.e. for `WHERE 100 = 100` or `100 = 101`, we would always do table scan
}
```

Reply:
Sure, I reoriented the logic similar to this and allowed getFirstColumnRefSide to return null so that it can be checked for, allowing me to get rid of the other method.